File: htsp/src/main/java/org/tvheadend/htsp/HtspConnectionInterface.java
package org.tvheadend.htsp;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

interface HtspConnectionInterface {
    void addMessageListener(@NonNull HtspMessageListener listener);

    void removeMessageListener(@NonNull HtspMessageListener listener);

    // synchronized, non blocking connect
    void openConnection();

    boolean isNotConnected();

    boolean isAuthenticated();

    // synchronized, blocking auth
    void authenticate();

    void sendMessage(@NonNull HtspMessage message);

    void sendMessage(@NonNull HtspMessage message, @Nullable HtspResponseListener listener);

    void closeConnection();
}

File: htsp/src/main/java/org/tvheadend/htsp/HtspConnectionInterface.kt
package org.tvheadend.htsp

internal interface HtspConnectionInterface {
    fun addMessageListener(listener: HtspMessageListener)
    fun removeMessageListener(listener: HtspMessageListener)

    // synchronized, non blocking connect
    fun openConnection()
    val isNotConnected: Boolean
    val isAuthenticated: Boolean

    // synchronized, blocking auth
    fun authenticate()
    fun sendMessage(message: HtspMessage)
    fun sendMessage(message: HtspMessage, listener: HtspResponseListener?)
    fun closeConnection()
}